backend/src/routes/songs.js

```diff
@@ -1,21 +1,40 @@
 const express = require('express');
 const geofire = require('geofire-common');
+const moment = require('moment');
 const { firebase } = require('../firebase-init.js');
 
 const db = firebase.firestore();
 
 async function handleSong(req, res) {
   const { lat, lng, songId } = req.body;
+  const currTime = moment().unix();
   if (!lat || !lng || !songId) {
     return res.status(400).send('Insufficient info');
   }
+
   // TODO: Add data validation
   const result = await db.collection('songEntries').add({
     songId,
     locationHash: geofire.geohashForLocation([lat, lng]),
     lat,
     lng,
+    ts: currTime,
   });
+
+  const userRef = db.collection('user_metadata').doc(req.user_email);
+  try {
+    await db.runTransaction(async (t) => {
```

Comment:
We're doing 1 read and 1 write here so idt it makes sense to use batch operation

Reply:
I think using .catch is a bit iffy b/c then you're returning res.status.send for the catch callback rather than the post callback. Gonna just stick to what I have